stixcore/tmtc/parameter.py

```diff
@@ -17,15 +21,68 @@ class Parameter:
     children : `list` optional
         Children of this parameter
     """
-    def __init__(self, name, value, idb_info, children=None):
+    def __init__(self, name, value, idb_info, *, children=None, order=-1):
         self.name = name
         self.value = value
         self.idb_info = idb_info
         self.children = children
+        self.order = order
+
+    def export(self, packetdata, exportstate, *, descr=False):
+
+        val = ''
+        convparam = getattr(packetdata, self.name)
+        if convparam:
+            if isinstance(convparam, EngineeringParameter):
+                val = convparam.engineering
+            elif isinstance(convparam, CompressedParameter):
+                val = convparam.decompressed
+
+            if isinstance(val, np.ndarray) and val.size > 1:
+                # idx = self.order-convparam.order-1
+                idx = exportstate[self.name]
+                exportstate[self.name] += 1
+                val = val[0][idx] if len(val.shape) > 1 else val[idx]
+
+            if self.name == 'NIX00445':
+                val = SCETime(coarse=self.value,
+                              fine=getattr(packetdata, "NIX00446",
+                                           Parameter('NIX00446', 0, None)).value)\
+                      .to_datetime().isoformat(timespec='milliseconds')
+            elif self.name == 'NIX00402':
+                if self.value > 2 ** 32 - 1:
+                    coarse = self.value >> 16
+                    fine = self.value & (1 << 16) - 1
+                else:
+                    coarse = self.value
+                    fine = 0
+                val = SCETime(coarse, fine).to_datetime().isoformat(timespec='milliseconds')
+            elif self.name == 'NIXD0003':
+                val = round(self.value / 2.5, 3)
```

Comment:
Is this to match Hualin's code or do we have something wrong?

Reply:
this is to match Hualin's format